docs/reference/index-modules/indexing-pressure.asciidoc

```diff
@@ -0,0 +1,60 @@
+[[index-modules-indexing-pressure]]
+== Indexing Pressure
+
+Indexing documents into Elasticsearch introduces system load in the form of
+memory and CPU load. Each indexing operation includes coordinating, primary, and
+replica stages. These stages can be performed across multiple nodes in the
+cluster. If too much indexing work is introduced into the system, the cluster
+can become saturated. This can adversely impact other components of the system
+such as searches, cluster coordination, background processing, etc.
+
+Indexing pressure is primarily generated by external operations such as indexing
+requests or internally by mechanisms such recoveries and cross-cluster
+replication.
+
+Elasticsearch internally monitors indexing load. When the load exceeds
+certain limits, new indexing work will be rejected.
+
+[float]
+=== Indexing Stages
+
+External indexing operations go through three stages: coordinating, primary, and
+replication. This write model is explain <<basic-write-model,here>>.
+
+[float]
+=== Memory Limits
+
+Elasticsearch exposes a node setting `indexing_pressure.memory.limit` which
+restricts the number of bytes for outstanding indexing requests. Be default,
+this setting is configured to be 10% of the heap.
+
+At the beginning of each <<indexing stage,here>>, Elasticsearch accounts for the
+bytes consumed by an indexing request. This accounting is only released at the
+end of the indexing stage. This means that upstream stages will account for the
+request overheard until all downstream stages are complete. For example, the
+coordinating request will remain accounted for until primary and replication
```

Comment:
I think that talking about "replication stage" is confusing (instead of replica stage). For me, the lifetime of replication starts when a primary completes the local indexing work, and completes when the primary has received an answer from all replicas, certainly not what's meant here.

Note that replication stage is used  confusingly throughout this doc